src/main.rs

```diff
@@ -10,6 +10,9 @@ async fn main() -> std::io::Result<()> {
     init_subscriber(subscriber);
 
     let settings = get_configuration().expect("Failed to read configuration");
-    let address = format!("{}:{}", settings.server.host, settings.server.port);
-    run(TcpListener::bind(address)?, settings)?.await
+    run(
+        TcpListener::bind((settings.server.host.as_str(), settings.server.port))?,
```

Comment:
It would be a bit more natural to use `&settings.server.host` rather than `settings.server.host.as_str()`

Reply:
`&settings.server.host` will actually be interpreted as `&String` not `&str` the former which does not implement the trait.

Reply:
Oh, and there's no auto coercion because we're putting it in a tuple. ok